### PR TITLE
Features/backend jwt authentication

### PR DESCRIPTION
**Pull request by neilscallywag**
This PR includes implementation of basic JWT authentcsation mechanism.

Changes:
Created Authentication Controller: The AuthController is responsible for handling login requests. It validates user credentials and issues JWTs for authenticated users.

Defined Models: We have created LoginModel and UserModel classes to handle user data during the login process.

JWT Generation: Upon successful authentication, a JWT is generated using the GenerateJSONWebToken method in the AuthController.

JWT Validation: The application now validates JWTs on protected endpoints. This is managed by the JwtBearer middleware configured in Startup.cs.

Error Handling: We have implemented custom error handling for missing JWT tokens and invalid credentials.

Added User Authentication Logic: For demonstration purposes, we have added a simple user validation logic. This will be replaced by actual user validation logic in future.

Blacklist Mechanism: Implemented a blacklist mechanism to revoke JWTs.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 .vs/
 bin/
 obj/
-/node_modules
\ No newline at end of file
+/node_modules
+backend/appsettings.json
```
Comment (ztdevelops) [inline .gitignore:6]: Why did you ignore `appsettings.json`? This file contains the connection string to the database.

File: backend/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*/appsettings.json
```
Comment (ztdevelops) [inline backend/.gitignore:1]: Double ignore, not required. Refer to [this comment](https://github.com/neilscallywag/ODR-ECORESPUTE/pull/21/files#r1207836040)).

File: .gitignore (hunk 1 above)
Comment (neilscallywag) [reply .gitignore:6]: It also contains JWT keys. Also the MYSQL connection credentials are not supposed to be pushed to github.

A blank boilerplate for appsettings.json is pushed with the name appsettings.development.json

File: backend/Controllers/AuthController.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            return new JwtSecurityTokenHandler().WriteToken(token);
+        }
+
+        private UserModel? AuthenticateUser(LoginModel login)
+        {
+            UserModel? user = null;
+
+            // Validate the User Credentials
+            // This is just an example, replace it with your actual user validation logic
+            if (login.Username == "admin" && login.Password == "passeword")
```
Comment (ztdevelops) [inline backend/Controllers/AuthController.cs:76]: Typo for password?

**Comment by ztdevelops** [comment]
Blacklist feature is not implemented yet.

https://github.com/neilscallywag/ODR-ECORESPUTE/pull/21/files#diff-6d20a08038168afcd55a1867e52912df06b52fb80097a48375b62e5415e065f6R110

File: .gitignore (hunk 1 above)
Comment (ztdevelops) [reply .gitignore:6]: `appsettings.json` should be the template file, whereas `appsettings.Development.json` should be the file with all the values, and should be the file that is ignored.

See [here](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0)).

Additionally, adding it to `.gitignore` does not remove the file from the project, but instead tells `git` to stop tracking from this point. For the purpose of this, the `appsettings.json` should be explicitly deleted.

In future, we can protect our configurations using [secrets](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows). No idea how to implement it, but we can figure it out later.

Comment (ztdevelops) [reply .gitignore:6]: > `appsettings.json` should be the template file, whereas `appsettings.Development.json` should be the file with all the values, and should be the file that is ignored.

Actually, scratch this part. I misunderstood the naming conventions.

**Review by ztdevelops — APPROVED** [review]

**Pull request merged**